tests/tools/s3_download.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python3
```

Comment:
Just a high level comment - Why not bring everything to this file? The Python code in the tests will use the API provided here and the rust code will use it as a `Command` ?  

This is more of a matter of opinion, but the other two files just have one class and one public function each, which all can be pulled to this file and the class can even be made private(ish) by prefixing it with `_` suggesting users are not advised to initialize this class themselves.

Reply:
I am not really experienced in Python, so this might be totally wrong :laughing: 

My requirements were:
- reuse the things from utils for both uploading and downloading resources. This means that `S3ResourceUploader` is also defined there and reuses some of the things that are currently available only for `download` (for example getting the latest version)
- the `S3ResourceUploader` is only used from python, so no need to have it as a script.
- the upload functionality is used from both python and rust so we need to have it as a script and as a function

What would be a good way to achieve this?

Reply:
The way I would have done is  - 

have only one file `s3_utils.py`  that defines 

1. `S3Wrapper` class and simple convenient APIs like `download_resources` (API if one wants to use it Everything from under `s3/__init__.py` and `s3/utils.py` ) and 
2. Under `if __name__ == '__main__':` define functionality that  can be called as a script. (things in `s3_download.py`) 
Except then the commands need to be explicit like `--download` and `--upload` (with options grouped suitably as needed.)

As a note:  In python when you `import <something>` part from the `if __name__ == '__main__':` does not get executed. So the same file can be used as a lib as well as a script.

Reply:
I am going to do this re-arrangement of files in a subsequent PR because it generates lots of merge conflicts.

Reply:
Yes - I agreed along with some other improvements to improvements :-)